File: weather-cli/weather.py
import sys
import requests

API_KEY = "e862e478cbd3791c0caf38a98f19abba"  # din OpenWeather key
BASE_URL = "https://api.openweathermap.org/data/2.5/weather"

def get_weather_openweather(city):
    params = {
        "q": city,
        "appid": API_KEY,
        "units": "metric",  
        "lang": "en"         
    }
    response = requests.get(BASE_URL, params=params)

    if response.status_code == 200:
        data = response.json()
        city_name = data["name"]
        temp = data["main"]["temp"]
        desc = data["weather"][0]["description"]
        return f"{city_name}: {temp}°C, {desc.capitalize()}"
    else:
        return None  


def get_weather_wttr(city):
    """Fallback til wttr.in"""
    url = f"http://wttr.in/{city}?format=3"
    response = requests.get(url)
    if response.status_code == 200:
        return response.text
    return "Could not fetch weather data from wttr.in either."


if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python weather.py <city>")
    else:
        city = " ".join(sys.argv[1:])  # støtter bynavn med mellomrom
        result = get_weather_openweather(city)

        if result:
            print(result)
        else:
            print("⚠️ OpenWeather not available, using fallback...")
            print(get_weather_wttr(city))

File: src/pages/api/weather.js
// pages/api/weather.js
export default async function handler(req, res) {
  const { city } = req.query;

  if (!city) {
    return res.status(400).json({ error: "City is required" });
  }

  const API_KEY = process.env.OPENWEATHER_API_KEY;
  const url = `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${API_KEY}&units=metric&lang=en`;

  try {
    const response = await fetch(url);
    const data = await response.json();

    if (response.ok) {
      return res.status(200).json({
        city: data.name,
        temp: data.main.temp,
        feels_like: data.main.feels_like,
        humidity: data.main.humidity,
        wind: data.wind.speed,
        desc: data.weather[0].description,
        source: "OpenWeather",
        raw: data, // 👈 legg med rå JSON for bevis
      });
    } else {
      // 👇 fallback til wttr.in
      const wttr = await fetch(`http://wttr.in/${city}?format=j1`);
      const wttrData = await wttr.json();
      const current = wttrData.current_condition[0];
      return res.status(200).json({
        city: city,
        temp: current.temp_C,
        feels_like: current.FeelsLikeC,
        humidity: current.humidity,
        wind: current.windspeedKmph,
        desc: current.weatherDesc[0].value,
        source: "wttr.in",
        raw: wttrData, // også med hele JSON her
      });
    }
  } catch (err) {
    return res.status(500).json({ error: "Failed to fetch weather data" });
  }
}
